trie/verkle.go

```diff
@@ -239,45 +237,31 @@ func nodeToDBKey(n verkle.VerkleNode) []byte {
 // Commit writes all nodes to the trie's memory database, tracking the internal
 // and external (for account tries) references.
 func (trie *VerkleTrie) Commit(onleaf LeafCallback) (common.Hash, int, error) {
```

Comment:
Here I'm accommodating to the new `BatchSerialize()` API. Intentionally, I avoid the `flush` channel style that we used since actually that wasn't giving concurrency gains.

The reason for that was that in `(n *InternalNode).Flush(..)` our first line was `n.Commit()` and then we recursively walked the trie sending nodes to the channel. But `n.Commit()` is actually the slow part, since it's updating the commitment from all nodes.

So that means that we do the slow part without sending anything to the channel, _and then_ we were walking and sending which mostly means sending "everything" quite fast. This didn't allow to "update nodes _while_ saving others to the database" (which I'd expect was the intention of the design).

Also, now that we're batching all serialization to avoid inverses, it doesn't make sense anymore to attempt doing this "update while saving" concurrency, since the idea of "batching" work means there's no concurrency (the _gains_ are coming from a cryptographic trick, not parallelism).